From 47155576403fea9339a1f702e3dc63455b763fb8 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Mon, 19 Apr 2010 20:40:34 -0400
Subject: [PATCH] Add session service activation file

Right now, if components of the desktop try to use
gnome-screensaver's dbus api before its started, then they
will fail.

This commit adds a service activation file, so dbus will start
it if it's needed and not running yet.

https://bugzilla.gnome.org/show_bug.cgi?id=616225
---
 configure.ac                          |    5 +++++
 data/Makefile.am                      |    4 ++++
 data/org.gnome.ScreenSaver.service.in |    3 +++
 3 files changed, 12 insertions(+), 0 deletions(-)
 create mode 100644 data/org.gnome.ScreenSaver.service.in

diff --git a/configure.ac b/configure.ac
index 3b933ce..6b40af5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -148,6 +148,10 @@ else
         AC_SUBST(DBUS_CFLAGS)
 fi
 
+# Find out where the session service file goes
+DBUS_SESSION_SERVICE_DIR=`pkg-config --variable session_bus_services_dir dbus-1`
+AC_SUBST(DBUS_SESSION_SERVICE_DIR)
+
 # Determine PAM prefix
 
 withval=""
@@ -1068,6 +1072,7 @@ src/gnome-screensaver.desktop.in
 data/Makefile
 data/gnome-screensavers.menu
 data/gnome-screensaver.pc
+data/org.gnome.ScreenSaver.service
 data/images/Makefile
 data/images/cosmos/Makefile
 savers/Makefile
diff --git a/data/Makefile.am b/data/Makefile.am
index 556e31d..fd9d8e5 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -43,6 +43,9 @@ gtkbuilder_DATA = 					\
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gnome-screensaver.pc
 
+dbussessionservicedir = $(DBUS_SESSION_SERVICE_DIR)
+dbussessionservice_DATA = org.gnome.ScreenSaver.service
+
 EXTRA_DIST = 				\
 	$(schemas_in_files)		\
 	$(desktop_in_files)		\
@@ -51,6 +54,7 @@ EXTRA_DIST = 				\
 	$(man_MANS)			\
 	gnome-screensaver		\
 	gnome-screensaver.pc.in		\
+	org.gnome.ScreenSaver.service.in\
 	xscreensaver-config.xsl		\
 	migrate-xscreensaver-config.sh	\
 	$(NULL)
diff --git a/data/org.gnome.ScreenSaver.service.in b/data/org.gnome.ScreenSaver.service.in
new file mode 100644
index 0000000..d46e4f4
--- /dev/null
+++ b/data/org.gnome.ScreenSaver.service.in
@@ -0,0 +1,3 @@
+[D-BUS Service]
+Name=org.gnome.ScreenSaver
+Exec=@EXPANDED_BINDIR@/gnome-screensaver --no-daemon
-- 
1.6.5.2

